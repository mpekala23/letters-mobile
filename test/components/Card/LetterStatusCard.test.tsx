import React from 'react';
import { LetterStatusCard } from '@components';
import { fireEvent, render, toJSON } from '@testing-library/react-native';

jest.mock('moment', () => () => ({
  format: () => 'Jul 12',
}));

const setup = (propOverrides = {}) => {
  const props = {
    status: 'Status',
    date: '2019-07-12',
    description: 'Description',
    color: 'green',
    onPress: jest.fn(),
    ...propOverrides,
  };
  return {
    ...render(<LetterStatusCard {...props} />),
    props,
  };
};

describe('Letter Status Card component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should fire onPress() on a press', () => {
    const { props, getByText } = setup();
    fireEvent.press(getByText('Status'));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should display status', () => {
    const { getByText } = setup();
    expect(getByText('Status')).toBeDefined();
  });

  it('should display date', () => {
    const { getByText } = setup();
    expect(getByText('Jul 12')).toBeDefined();
  });

  it('should display description', () => {
    const { getByText } = setup();
    expect(getByText('Description')).toBeDefined();
  });
});
