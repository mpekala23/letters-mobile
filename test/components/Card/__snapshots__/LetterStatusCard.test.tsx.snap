// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter Status Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
    testID="letterStatusCard"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "height": "100%",
              "marginRight": 15,
              "width": 6,
            },
            Object {
              "backgroundColor": "",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins-Bold",
                },
                Object {
                  "color": "#282828",
                  "fontSize": 20,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            Status
          </Text>
          <Text
            style={
              Object {
                "color": "#6E757C",
                "fontSize": 16,
              }
            }
          >
            Jul 12
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#6E757C",
              "fontSize": 16,
            }
          }
        >
          Description
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
